lib/shared/puppet_self_service.rb

```diff
@@ -122,3 +122,25 @@ def self.filesystem_free(path)
     (stat.blocks_available.to_f / stat.blocks.to_f * 100).to_i
   end
 end
+
+def psql_return_result(sql, psql_options = '')
+  command = %(su pe-postgres --shell /bin/bash --command "cd /tmp && #{PUP_PATHS[:server_bin]}/psql #{psql_options} --command \\"#{sql}\\"")
+  Facter::Core::Execution.execute(command)
+end
+
+def self.max_connections
+  sql = %Q(
+    select count(*) used from pg_stat_activity;
+  )
+  psql_options = '-qtAX'
+  psql_return_result(sql, psql_options)
+end
+
+
+def self.cur_connections
+  sql = %Q(
+    select count(*) used from pg_stat_activity;
```

Comment:
Selecting "active" connections would need to have `where state = 'active'`, otherwise we'd also pick up connections that have already finished and are in `idle` state.

```
select count(*) used from pg_stat_activity where state = 'active';
```